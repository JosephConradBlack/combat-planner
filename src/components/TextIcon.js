import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Item, Icon } from 'semantic-ui-react';

const IconContainer = styled(Item.Extra)`
  &&& {
    position: relative;
    height: 33px;
    width: 33px;
  }
`;

const StyledIcon = styled(Icon)`
  &&& {
    position: absolute;
    left: 0;
  }
`;

const StyledText = styled.div`
  &&& {
    position: absolute;
    top: 3px;
    left: 40%;
    color: ${props => props.color};
  }
`;

const TextIcon = ({ icon, number, color }) => {
  return (
    <div>
      <IconContainer>
        <StyledIcon size="big" name={icon} color={color} />
        <StyledText color={color}>{number}</StyledText>
      </IconContainer>
    </div>
  );
};

TextIcon.propTypes = {
  icon: PropTypes.string,
  number: PropTypes.number,
  color: PropTypes.string
};

TextIcon.defaultProps = {
  icon: 'heart outline',
  number: 0,
  color: 'red'
};

export default TextIcon;
