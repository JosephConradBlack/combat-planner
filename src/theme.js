export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
