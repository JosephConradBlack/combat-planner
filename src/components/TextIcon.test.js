import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import TextIcon from './TextIcon';

describe('TextIcon', () => {
  it('renders with no props', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <TextIcon />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with icon', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <TextIcon icon="shield" />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with number', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <TextIcon number={4} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with color', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <TextIcon color={'blue'} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
