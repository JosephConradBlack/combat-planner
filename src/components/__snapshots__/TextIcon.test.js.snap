// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextIcon renders with color 1`] = `
<div>
  <div
    className="extra sc-bdVaJa cdKNHG"
  >
    <i
      aria-hidden="true"
      className="blue heart outline big icon sc-bwzfXH cXPuzX"
    />
    <div
      className="sc-htpNat gnxApo"
      color="blue"
    >
      0
    </div>
  </div>
</div>
`;

exports[`TextIcon renders with icon 1`] = `
<div>
  <div
    className="extra sc-bdVaJa cdKNHG"
  >
    <i
      aria-hidden="true"
      className="red shield big icon sc-bwzfXH cXPuzX"
    />
    <div
      className="sc-htpNat iPQVHr"
      color="red"
    >
      0
    </div>
  </div>
</div>
`;

exports[`TextIcon renders with no props 1`] = `
<div>
  <div
    className="extra sc-bdVaJa cdKNHG"
  >
    <i
      aria-hidden="true"
      className="red heart outline big icon sc-bwzfXH cXPuzX"
    />
    <div
      className="sc-htpNat iPQVHr"
      color="red"
    >
      0
    </div>
  </div>
</div>
`;

exports[`TextIcon renders with number 1`] = `
<div>
  <div
    className="extra sc-bdVaJa cdKNHG"
  >
    <i
      aria-hidden="true"
      className="red heart outline big icon sc-bwzfXH cXPuzX"
    />
    <div
      className="sc-htpNat iPQVHr"
      color="red"
    >
      4
    </div>
  </div>
</div>
`;
